packages/@aws-cdk/integ-tests/lib/assertions/private/deploy-assert.ts

```diff
@@ -42,10 +50,10 @@ export class DeployAssert extends Construct implements IDeployAssert {
 
   public scope: Stack;
 
-  constructor(scope: Construct) {
+  constructor(scope: Construct, props?: DeployAssertProps) {
     super(scope, 'Default');
 
-    this.scope = new Stack(scope, 'DeployAssert');
+    this.scope = new Stack(scope, 'DeployAssert', { env: props?.env });
```

Comment:
I think it might be better to actually allow the user to pass their own `Stack`. I think that is
what I originally intended to do. That also opens up the posibility to put the assertions in the
same Stack as your test stack.

Reply:
I like that a lot more! Let me refactor this PR

I also think it's beneficial in the case the user wanting to keep the stacks separated. It allows for creating lambda functions for the Lambda assertions independent from what is being integration tested